openapi_python_client/parser/properties/int.py

```diff
@@ -60,10 +60,13 @@ def convert_value(cls, value: Any) -> Value | None | PropertyError:
             return value
         if isinstance(value, str):
             try:
-                int(value)
+                value = float(value)
             except ValueError:
                 return PropertyError(f"Invalid int value: {value}")
-            return Value(value)
+        if isinstance(value, float):
+            as_int = int(value)
+            if value == as_int:
+                value = as_int
```

Comment:
The changes here and on line 63 are to support the "merge int with float" behavior in `merge_properties.py`. Say that you have a float property with a default of 2. Even if it was written as just `2` in the spec file, when it gets converted to a float and then to a string and then to a Value, we end up with `Value("2.0")`. That is a perfectly reasonable integer value, but Python won't actually let you do `int("2.0")`; you have to parse it as a float first, and then verify that it has no fractional part.

Reply:
This wasn't caught by my unit tests previously because `float_property_factory` was skipping the normal logic for converting the default to a Value, and also in the end-to-end tests we didn't happen to have the int & float with float default case.